nvflare/apis/study_manager_spec.py

```diff
@@ -12,40 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import datetime
+from datetime import datetime
+
+from .fl_context import FLContext
 
 
 class Study:
     def __init__(
         self,
         name: str,
         description: str,
-        sites: [str],
-        users: [str],
-        start_time: datetime.datetime,
-        end_time: datetime.datetime,
+        contact: str,
+        participating_clients: [str],
+        participating_admins: [str],
+        start_date: datetime.date,
+        end_date: datetime.date,
         reviewers=None,
     ):
         self.name = name
         self.description = description
-        self.sites = sites
-        self.users = users
-        self.start_time = start_time
-        self.end_time = end_time
+        self.contact = contact
+        self.participating_clients = participating_clients
+        self.participating_admins = participating_admins
+        self.start_date = start_date
+        self.end_date = end_date
         self.reviewers = reviewers
-        self.create_time = None
+        self.created_at = datetime.utcnow().isoformat()
 
 
 class StudyManagerSpec(object):
```

Comment:
we should make this class an abstract class and its method abstractmethod so who inherits it need to implement those methods.

Reply:
This study is concrete, not a spec, I believe.  The study itself is like a dataclass.  It does not have methods to implement.  Also, the __init__ also means the class here is not a spec.

Reply:
Thanks, I agree study is a dataclass.
What about "StudyManagerSpec"?

Reply:
Now it is an ABC with all methods as abstractmethod.